### PR TITLE
Feature/all types of charts

### PR DESCRIPTION
**Pull request by francozacco**

### CODE REVIEW

File: api/client/dashboard_assetlist_chart.go [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+	items := map[string]DashboardAssetListChart{}
+	err = json.NewDecoder(body).Decode(&items)
+	if err != nil {
+		return nil, err
+	}
+	return &items, nil
+}
+
+func (c *Client) CreateDashboardAssetListChart(item *DashboardAssetListChartParams) (*DashboardAssetListChart, error) {
+	buf := bytes.Buffer{}
```
Comment (smartinellimarco) [inline api/client/dashboard_assetlist_chart.go:36]: perhaps we should make the clients take the bytes as input and reuse them

**Review by marianogaragiola — APPROVED** [review]

File: examples/resources/splight_dashboard_table_chart/resource.tf [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+      name = "query filter asset name"
+    }
+    query_filter_attribute {
+      id   = "1234-1234-1234-1234"
+      name = "query filter attribute name"
+    }
+    query_plain = jsonencode([
+      {
+        "$match" = {
+          asset     = "1234-1234-1234-1234"
```
Comment (smartinellimarco) [inline examples/resources/splight_dashboard_table_chart/resource.tf]: use the references

**Review by smartinellimarco — CHANGES_REQUESTED** [review]

File: api/client/dashboard_assetlist_chart.go (hunk 1 above)
Comment (francozacco) [reply api/client/dashboard_assetlist_chart.go:36]: yeah I did something like that before tho the amount of code we reuse is not reducing the number of lines too much so I decided to leave it like it is for now.

File: examples/resources/splight_dashboard_table_chart/resource.tf (hunk 2 above)
Comment (francozacco) [reply examples/resources/splight_dashboard_table_chart/resource.tf]: Creating the asset and attributes too is like too much boilerplate for the example.. tho I can do it if you want.

**Review by smartinellimarco — APPROVED** [review]

**Pull request merged**